### PR TITLE
Upgrade Account contract

### PR DESCRIPTION
**Pull request by microbecode**
- Change Account contract to the newest OpenZeppelin version. Not bringing in all of the OZ files, just hand-picked the required ones to one folder and fixed their imports
- Non-account related contracts needed to be moved to their own folder, because compilation has to be done separately for account contracts and non-account contracts
- Updated compilation script to reflect the changes

### CODE REVIEW

**Review by JaniAnttonen — COMMENTED** [review]
No tests? Would love to see the contract in use.

File: contracts/contracts/account/constants.cairo [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const UINT8_MAX = 256
+
+#
+# Interface Ids
+#
+
+# ERC165
+const IERC165_ID = 0x01ffc9a7
+const INVALID_ID = 0xffffffff
```
[the remark this answers is not in the document]
Comment (microbecode) [reply contracts/contracts/account/constants.cairo:25]: This code is from OpenZeppelin, so I better not touch it.

**Comment by microbecode** [comment]
> No tests? Would love to see the contract in use.

This version doesn't utilize the account contract explicitly. Because of the version bump of the used libraries, the existing unit tests know automatically how to call the new version of the account contract. So all old unit tests are still valid. Basically just the account contract structure has changed a bit, but because we also updated library versions, the newer versions know to call the new structure.

**Comment by microbecode** [comment]
oops, wrong button, didn't mean to close the PR :)

**Comment by JaniAnttonen** [comment]
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>1b0e06e</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://starknet-multisig-fuezgmvki-equilibriumco.vercel.app'>https://starknet-multisig-fuezgmvki-equilibriumco.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/equilibriumco/starknet-multisig/CEPNAW8CDEvgDBg7Z4EozcNEA8VL'>https://vercel.com/equilibriumco/starknet-multisig/CEPNAW8CDEvgDBg7Z4EozcNEA8VL</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/eqlabs/starknet-multisig/actions/runs/2292059215)

**Pull request merged**